parsers/parse-clang-diagnostic-groups.py

```diff
@@ -386,6 +387,20 @@ def get_messages(self, enabled_by_default: bool) -> list[str]:
 
         return [diag.text for diag in self.diagnostics]
 
+    def is_remark(self) -> bool:
+        """
+        Return whether a group is a remark.
+
+        :return: True if the group has remark diagnostics, False otherwise.
+        """
+        rem=False
+
+        for child in self.diagnostics:
+            if child.is_remark:
+                rem=True
+
```

Comment:
[LLVM's ClangDiagnosticsEmitter](https://github.com/llvm/llvm-project/blob/llvmorg-13.0.0/clang/utils/TableGen/ClangDiagnosticsEmitter.cpp) looks recursively at subgroups; this function needs to do the same (`self.children`).

Also, ClangDiagnosticsEmitter raises an error if a group has both remark and non-remark messages. It would be helpful to raise an exception if this assumption is violated, as it means we need to update this parser.